src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +441,50 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+/// <summary>
+/// Set the CustomHttpHeaders property from the custom http headers string.
+/// </summary>
+/// <param name="stringValue">The value to set.</param>
+public void ValidateAndSetCustomHttpHeadersFromString(string headersValue) 
+{
+    if (String.IsNullOrWhiteSpace(headersValue))
+    {
+        return;
+    }
+
+    this.CustomHttpHeaders = new List<string>();
+
+    string[] headerElements = headersValue.Split(new string[] { Environment.NewLine },StringSplitOptions.None);
+    foreach (var headerElement in headerElements)
+    {
+        // Trim header for empty spaces
+        var header = headerElement.Trim();
+
+        // A header string must have atleast one semi-colon
+        if (GetSemiColons(header) > 0)
+        {
+            CustomHttpHeaders.Add(header);
+        }
+        else
+        {
+            throw new ArgumentException("A http header string must have a semi-colon delimeter");
+        }
+    }
+
+}
+
+private static int GetSemiColons(string input)

```

Comment:
The name does not indicate what this method actually does, GetSemiColonCount would probably be a more appropriate name.

Reply:
I have removed the entire GetSemiColons method